testsuite/features/core_proxy_branch_network.feature

```diff
@@ -0,0 +1,140 @@
+# Copyright (c) 2018 SUSE LLC
+# Licensed under the terms of the MIT license.
+#
+# The scenarios in this feature are skipped if there is no proxy
+# ($proxy is nil) or if there is no private network ($private_net is false)
+
+Feature: Setup SUSE Manager for Retail branch network
+  In order to deploy SUSE Manager for Retail solution
+  As the system administrator
+  I want to prepare the branch network
+
+@proxy
+@private_net
+  Scenario: Install or update branch network formulas on the server
+    When I manually install the "branch-network" formula on the server
+    And I manually install the "dhcpd" formula on the server
+    And I manually install the "bind" formula on the server
+    And I wait for "16" seconds
```

Comment:
That gonna wait until the command is finished?
```
When(/^I manually install the "([^"]*)" formula on the server$/) do |package|
  $server.run("zypper --non-interactive install --force #{package}-formula")
end
```
If that's the case, we can avoid the unconditional wait step?
If not, can we add some check and wait inside the installation step, so we are sure we can proceed with next step?

Reply:
Btw, looks weird for me to have only "When" in that test case, bit breaking the natural language of Gherkin.
At least, When and Then? :p

Reply:
@srbarrios Good remark (wait for 16 seconds). Here it how it goes:
 * you install the package on the server
 * then the formula becomes available in the UI

I don't know why it does not become available immediately. I would prefer too to use a wait until it is available in the UI, or better, until whatever happens behind the scenes and that is needed for it to appear in the UI is finished.

I admit I did the sleep to get this solved quickly, but it's suboptimal, like any other unconditional wait. Feel free to fix if you find a solution.

Reply:
@srbarrios No, `When` is correct here, it is an action, not a test. Installing formulas is a typical installation action, a precondition before a test.

If you look more in detail, I also try to respect the `When / Then` structure of gherkin. But `Then`s should really be tests, with a `should`, not fake 'Then`s just for the sake of the sequence.

What we allow ourselves is several `When / Then` sequences in same scenario.

Reply:
Imho the last And, with the wait statement, it should be a Then checking if the previous steps are doing what we want. For instance, checking if those are available in the UI.
Otherwise, can't be promoted as a Test Case, it's more... like a pre-condition for other tests.

Reply:
It is. Feel free to reorganize if you want. But please don't change something that is intrinsically an action to a test just for the sake of the order of the keywords.